common/src/main/scala/uk/ac/wellcome/utils/TryBackoff.scala

```diff
@@ -32,9 +33,12 @@ trait TryBackoff extends Logging {
       if (attempt > 0) increaseWaitTimeExponentially(attempt)
       else baseWaitMillis
 
-    val cancellable = system.scheduler.scheduleOnce(waitTime milliseconds)(
-      run(f, system, numberOfAttempts))
-    maybeCancellable = Some(cancellable)
+    if (continuous && !succeeded) {
```

Comment:
This condition is still wrong, I think.

If it was continuous _or_ we failed, we should retry.